src/Microsoft.Management.Configuration/ConfigurationSetParser.cpp

```diff
@@ -86,15 +105,60 @@ namespace winrt::Microsoft::Management::Configuration::implementation
         {
             return std::make_unique<ConfigurationSetParser_0_1>(std::move(document));
         }
+        else if (schemaVersion.PartAt(0).Integer == 0 && schemaVersion.PartAt(1).Integer == 2)
+        {
+            return std::make_unique<ConfigurationSetParser_0_2>(std::move(document));
+        }
+
+        AICLI_LOG(Config, Error, << "Unknown configuration version: " << schemaVersion.ToString());
+        return std::make_unique<ConfigurationSetParserError>(WINGET_CONFIG_ERROR_UNKNOWN_CONFIGURATION_FILE_VERSION, GetFieldName(FieldName::ConfigurationVersion), versionNode.as<std::string>());
+    }
+
+    bool ConfigurationSetParser::IsRecognizedSchemaVersion(hstring value) try

```

Comment:
maybe there can be something like an enum that has the recognized versions, unknown and max. a helper method that returns the string or semantic representation and compare with the value given. then Create and current callers of IsRecognizedSchemaVersion throw if the result is unknown and LatestVersion just uses max. but maybe is too much.

Reply:
I don't think an enum is warranted here, and I would only expect the value to be set in the event that the caller knows what they are doing with it (and therefore knows the versions).